### PR TITLE
docs: add docs for ClusterOperator for operators

### PR DESCRIPTION
**Pull request by abhinavdahiya**
/cc @crawford

### CODE REVIEW

File: docs/dev/clusteroperator.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# ClusterOperator Custom Resource
+
+The ClusterOperator is a custom resource object which holds the current state of an operator. This object is used by operators to convey their state to the rest of the cluster.
+
+Ref: [godoc](https://godoc.org/github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1#ClusterOperator) for more info on the ClusterOperator type.
+
+## Why I want ClusterOperator Custom Resource in /manifests
+
+ClusterVersionOperator sweeps the release payload and apply it to the cluster.
```
Comment (wking) [inline docs/dev/clusteroperator.md]: nit: "apply" -> "applies".  Also, this sentence leads into the next; they should probably go in the same paragraph.

File: docs/dev/clusteroperator.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+So if your operator manages a critical component for the cluster and you want ClusterVersionOperator to wait for your operator to **complete** before it continues to apply other operators, you must include the ClusterOperator Custom Resource in [`/manifests`](./operators.md#what-do-i-put-in-manifests).
```
Comment (wking) [inline docs/dev/clusteroperator.md]: nit: no need for the `./` here.

File: docs/dev/clusteroperator.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  name: my-cluster-operator
+status:
+  version: 1.0.0
+```
+
+## What should an operator report with ClusterOperator Custom Resource
+
+### Status
+
+The operator should ensure that all the fields of `.status` in ClusterOperator are atomic changes. This means that all the fields in the `.status` are only valid together and not partially represent the status of the operator.
```
Comment (wking) [inline docs/dev/clusteroperator.md]: nit: "not" -> "do not"

File: docs/dev/clusteroperator.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+### Version
+
+The operator should report a version which indicates the components that it is applying to the cluster.
+
+### Conditions
+
+[Refer](https://godoc.org/github.com/openshift/cluster-version-operator/pkg/apis/operatorstatus.openshift.io/v1#ClusterStatusConditionType) the godocs for conditions.
```
Comment (wking) [inline docs/dev/clusteroperator.md]: nit: The link should be from `the godocs`, not `Refer`.

File: docs/dev/operators.md [hunk 5]
```diff
@@ -18,13 +20,13 @@ You need the following:
...
-  - A service account and a service account role binding
-  - Deployment for your operator
-  - An OperatorStatus CR
-  - Any other config objects your operator might need
-  - An image-references file (See below)
+- Namespace for your operator
+- Roles your operator needs
+- A service account and a service account role binding
+- Deployment for your operator
+- An ClusterOperator CR [more info here](./clusteroperator.md)
```
Comment (wking) [inline docs/dev/operators.md]: nit: "An" -> "A".  And you can drop the `./`.

**Comment by abhinavdahiya** [comment]
@wking updated to match the comments.

**Comment by abhinavdahiya** [comment]
The bot didn't pick up the `/lgtm` :P

**Pull request merged**